xmr-btc/src/alice.rs

```diff
@@ -2,33 +2,178 @@ use anyhow::{anyhow, Result};
 use ecdsa_fun::adaptor::{Adaptor, EncryptedSignature};
 use rand::{CryptoRng, RngCore};
 
-use crate::{bitcoin, bitcoin::GetRawTransaction, bob, monero, monero::ImportOutput};
-use ecdsa_fun::{nonce::Deterministic, Signature};
+use crate::{
+    bitcoin,
+    bitcoin::{BroadcastSignedTransaction, GetRawTransaction},
+    bob, monero,
+    monero::{ImportOutput, Transfer},
+    transport::SendReceive,
+};
+use ecdsa_fun::nonce::Deterministic;
 use sha2::Sha256;
+use std::convert::{TryFrom, TryInto};
+
+pub mod message;
+pub use message::{Message, Message0, Message1, Message2, UnexpectedMessage};
+
+pub async fn next_state<
```

Comment:
:question: How do you see these `next_state` functions fitting in with a function that orchestrates the swap for Alice or Bob?

I see the benefit of this design as allowing you to control how far the state machine goes, which is useful for testing, but I think the end user of this library will want the swap to be executed to completion and for all the abort scenarios to be handled for him.

It would be nice if we could find a way to keep the flexibility for testing and provide such a high-level API at the same time.

Reply:
In that case you can simple ask to run to the last state which will do what you are asking. This is why I kept the run_until() functions in the test lib. For production we would use a similar function except it wont take in a desired end state will always try and run until the end state.
I am not sure how the abort scenarios will tie in. I think it is probably best to write a test or design a client so we can get a better idea how the API should be.

Reply:
> I think it is probably best to write a test or design a client so we can get a better idea how the API should be.

That's a fair point.

I guess for the time being it is sufficient if we offer the building blocks to handle all the abort scenarios, and if we develop an application that uses this library we can continue to iterate on it.